src/gui_common/CustomRichTextLabel.cs

```diff
@@ -207,43 +219,71 @@ private void ParseCustomTags()
     ///   Returns a templated bbcode string for the given custom tag.
     /// </summary>
     /// <param name="input">The string tagged by custom tags</param>
-    /// <param name="tag">Custom Thrive bbcode-styled tags</param>
-    private string BuildTemplateForTag(string input, ThriveBbCodeTag tag)
+    /// <param name="bbcode">Custom Thrive bbcode-styled tags</param>
+    /// <param name="attributes">Attributes specifying an additional functionality to the bbcode.</param>
+    private string BuildTemplateForTag(string input, ThriveBbCode bbcode, string[] attributes = null)
     {
         // Defaults to input so if something fails output returns unchanged
         var output = input;
 
-        switch (tag)
+        switch (bbcode)
         {
-            case ThriveBbCodeTag.Compound:
+            case ThriveBbCode.Compound:
             {
-                if (SimulationParameters.Instance.DoesCompoundExist(input))
+                if (!SimulationParameters.Instance.DoesCompoundExist(input))
                 {
-                    var compound = SimulationParameters.Instance.GetCompound(input);
-
-                    output = $"[b]{compound.Name}[/b] [font=res://src/gui_common/fonts/" +
-                        $"BBCode-Image-VerticalCenterAlign-3.tres] [img=20]{compound.IconPath}[/img][/font]";
+                    GD.Print($"Compound: \"{input}\" doesn't exist, referenced in bbcode");
+                    break;
                 }
-                else
+
+                var compound = SimulationParameters.Instance.GetCompound(input);
+
+                var name = compound.Name;
+
+                // Parse attributes if there is any
+                // ReSharper disable MergeSequentialChecksWhenPossible
+                if (attributes != null && attributes.Length > 0)
                 {
-                    GD.Print($"Compound: \"{input}\" doesn't exist, referenced in bbcode");
+                    if (attributes[0].BeginsWith("text="))
+                    {
+                        var split = attributes[0].Split("=");
```

Comment:
I think this would be nice to split to a general parsing method to convert list of "key=value" pairs into a Dictionary<string,string>.